packages/communication-react/review/communication-react.api.md

```diff
@@ -1647,27 +1647,53 @@ export interface MessageThreadStyles extends BaseCustomStyles {
 // @public
 export const MicrophoneButton: (props: MicrophoneButtonProps) => JSX.Element;
 
+// @public
+export interface MicrophoneButtonContextualMenuStyles extends IContextualMenuStyles {
+    menuItemStyles?: IContextualMenuItemStyles;
+}
+
 // @public
 export interface MicrophoneButtonProps extends ControlBarButtonProps {
+    microphones?: OptionsDevice[];
+    onSelectMicrophone?: (device: OptionsDevice) => Promise<void>;
+    onSelectSpeaker?: (device: OptionsDevice) => Promise<void>;
     onToggleMicrophone?: () => Promise<void>;
+    selectedMicrophone?: OptionsDevice;
+    selectedSpeaker?: OptionsDevice;
+    showDeviceSelectionMenu?: boolean;
+    speakers?: OptionsDevice[];
     strings?: Partial<MicrophoneButtonStrings>;
+    styles?: Partial<MicrophoneButtonStyles>;
 }
 
 // @public
 export type MicrophoneButtonSelector = (state: CallClientState, props: CallingBaseSelectorProps) => {
     disabled: boolean;
     checked: boolean;
+    microphones: AudioDeviceInfo[];
+    speakers: AudioDeviceInfo[];
+    selectedMicrophone?: AudioDeviceInfo;
+    selectedSpeaker?: AudioDeviceInfo;
 };
 
 // @public
 export interface MicrophoneButtonStrings {
+    microphoneMenuTitle?: string;
+    microphoneMenuTooltip?: string;
     offLabel: string;
     onLabel: string;
+    speakerMenuTitle?: string;
+    speakerMenuTooltip?: string;
     tooltipDisabledContent?: string;
     tooltipOffContent?: string;
     tooltipOnContent?: string;
 }
```

Comment:
Not related to this ticket but I created a [ticket](https://skype.visualstudio.com/SPOOL/_workitems/edit/2713109) to make all string properties consistent

Reply:
We'll have to make them all optional. We can only add optional fields to existing types.